### PR TITLE
:recycle: loop over constituents, then open boundaries

### PR DESCRIPTION
**Pull request by SanderHulst**
See issue #171

### CODE REVIEW

File: adcircpy/fort15.py [hunk 1]
```diff
@@ -606,10 +606,10 @@ def get_tidal_forcing(self) -> str:
                         # f'{len(self.mesh.open_boundaries)}',
                     ]
                 )
-            for index, row in self.mesh.boundaries.ocean.gdf.iterrows():
                 for constituent in self.mesh.forcings.tides.get_active_constituents():
                     f.append(fort15_line(constituent))
-                    vertices = self.mesh.get_xy(crs='EPSG:4326').iloc[row.indexes, :].values
+                for index, row in self.mesh.boundaries.ocean.gdf.iterrows():
+                    vertices = self.mesh.get_xy(crs='EPSG:4326').loc[row.indexes, :].values
```
Comment (SorooshMani-NOAA) [inline adcircpy/fort15.py]: Why was `iloc` changed to `loc`? Are the position of *xy* rows the same as the dataframe index?

File: adcircpy/fort15.py [hunk 2]
```diff
@@ -606,10 +606,10 @@ def get_tidal_forcing(self) -> str:
                         # f'{len(self.mesh.open_boundaries)}',
                     ]
                 )
-            for index, row in self.mesh.boundaries.ocean.gdf.iterrows():
```
Comment (SorooshMani-NOAA) [inline adcircpy/fort15.py:609]: I'm not sure if this change necessarily makes the fort15 format correct based on the ADCIRC documentation. This seems to mess up with the order of the two separate loops on constituents in the ADCIRC documentation.


I see the issue with the old code, but I don't think the updated code in this PR looks right either!
Maybe you meant to also de-indent the `for constituent in ...` loop?! 


https://adcirc.org/home/documentation/users-manual-v51/input-file-descriptions/model-parameter-and-periodic-boundary-condition-file-fort-15/#:~:text=end%20k%20loop-,NBFR,end%20k%20loop,-ANGINN

image

**Review by SorooshMani-NOAA — CHANGES_REQUESTED** [review]
Although I see the issue with the old code based on the documentation, I believe the changes introduce new issues. I'm not an ADCIRC user, so I'd like @WPringle to also comment on this if he has time. Thanks!

File: adcircpy/fort15.py (hunk 1 above)
Comment (SanderHulst) [reply adcircpy/fort15.py]: That is accidental. I copied lines from 1.2.2. Thanks!

File: adcircpy/fort15.py (hunk 2 above)
Comment (SanderHulst) [reply adcircpy/fort15.py:609]: ahum, trying a new version 😊

**Review by WPringle — COMMENTED** [review]
It looks like the new code does make sense based on how ADCIRC fort.15 is constructed. It seems the old code only works with one open boundary perhaps?

**Comment by WPringle** [comment]
@SorooshMani-NOAA Do we have some unit tests to show passing with the updates?

**Comment by SanderHulst** [comment]
It seems it only worked for one open boundary, yes.

**Comment by SorooshMani-NOAA** [comment]
> @SorooshMani-NOAA Do we have some unit tests to show passing with the updates?

@WPringle there are tests, but I'm not sure how much of the code is being tested. I was not very involved with `adcircpy` development. In any case I'll check if there are any tests that need to be updated due to this.

**Review by SorooshMani-NOAA — APPROVED** [review]

**Comment by WPringle** [comment]
@SorooshMani-NOAA Did tests run on your end? As they seem fail on here aside from linting.

**Comment by SorooshMani-NOAA** [comment]
> @SorooshMani-NOAA Did tests run on your end? As they seem fail on here aside from linting.

I'm trying to figure out what's going on. I don't know why it fails!

**Comment by SorooshMani-NOAA** [comment]
Running test locally shows a test that fails due to updated `fort15` order. I'm going to update that reference. But I'm still not sure why the GitHub test fails. I might need to test limiting version to Python 3.10 and see if it matters!

**Comment by SorooshMani-NOAA** [comment]
@SanderHulst @WPringle two of the tests need fixing (unrelated to this PR) and then also the tests fail on python 3.11 because `netcdf4` version `<1.6` doesn't work on python `3.11` or something like that! So I'm fixing these issues in a separate branch to merge to main, then you can merge main to your branch and tests should run.

**Update:**
Additionally I'd like to add that there's no test failure due to this update, so I think it would be a good idea to include a basic test case that demonstrate the fix in this PR if possible @SanderHulst 

**Update 2:**
Merged #173

**Comment by SorooshMani-NOAA** [comment]
@SanderHulst you can merge main to this branch so that we can rerun the tests

**Pull request merged**